### PR TITLE
PLU-254, PLU-265: add customisations for flow step

### PR DESCRIPTION
**Pull request by m0nggh**
## Problem

The flow step content is not too intuitive and cannot be customised.
image

## Solution

1. Display event name instead of app name to give more context to users, this will be the default state in the future, in which we will be implementing functionality to edit the step name.
2. Each step's accordion label should be different, only `ChooseConnectionSubstep` and `FlowSubstep` require customisable labels now.

Updated example:
image

## Tests
- [x] Existing pipes still work
- [x] If no label is given for the connection step or settings step, fallback to default states

### CODE REVIEW

File: packages/types/index.d.ts [hunk 1]
```diff
@@ -408,6 +408,8 @@ export interface IApp {
   connections?: IConnection[]
   description?: string
   isNewApp?: boolean
+  connectionStepLabel?: string // for step accordion label
+  settingsStepLabel?: string // for step accordion label: app level
```
Comment (ogp-weeloong) [inline packages/types/index.d.ts:412]: nit: how about this so that we group everything together:
```suggestion
  substepLabels: {
    connectionSubstep: string
    setupSubsetup: string
  }
```

File: packages/frontend/src/graphql/queries/get-apps.ts [hunk 2]
```diff
@@ -20,6 +20,8 @@ export const GET_APPS = gql`
       connectionCount
       description
       isNewApp
+      connectionStepLabel
```
Comment (ogp-weeloong) [inline packages/frontend/src/graphql/queries/get-apps.ts:23]: not sure if i'm missing something, but you're already replacing the substep name in the backend. why is this needed?

Comment (ogp-weeloong) [reply packages/frontend/src/graphql/queries/get-apps.ts:23]: Discussed - this is to support different labels depending on the event chosen

File: packages/backend/src/helpers/get-app.ts [hunk 3]
```diff
@@ -40,18 +40,18 @@ const getApp = (appKey: string, stripFuncs = true) => {
   return appData
 }
 
-const chooseConnectionStep = {
-  key: 'chooseConnection',
-  name: 'Choose connection',
-}
-
-const testStep = (stepType: 'trigger' | 'action') => {
+const chooseConnectionStep = (connectionStepLabel?: string) => {
```
Comment (ogp-weeloong) [inline packages/backend/src/helpers/get-app.ts]: i think we can omit the backend changes, since the frontend always needs to account for per-action/per-trigger labels. So this is kinda duplicating logic and spreads complexity across both backend and frontend.

I would instead consolidate all the complexity inside the frontend by making it handle all these label changes. For example, I could change `FlowStep` to something like this:
```
substeps.map((substep: ISubstep, index: number) => (
  <Fragment key={`${substep?.name}-${index}`}>
    {substep.key === 'chooseConnection' && app && (
      <ChooseConnectionSubstep
        /** props omitted for brevity **/
        label={app.connectionSetupLabel ?? substep.name}
      />
    )}
    {substep.key === 'testStep' && (
      <TestSubstep
        /** props omitted for brevity **/
      />
    )}
    {substep.key &&
      ['chooseConnection', 'testStep'].includes(substep.key) ===
        false && (
        <FlowSubstep
          /** props omitted for brevity **/
          settingsLabel={selectedActionOrTrigger?.settingsStepLabel ?? app.settingsStepLabel ?? substep.name}
        />
      )}
  </Fragment>
)
```

**Review by ogp-weeloong — CHANGES_REQUESTED** [review]
lmk what you think of my backend comment?

File: packages/backend/src/helpers/get-app.ts (hunk 3 above)
Comment (m0nggh) [reply packages/backend/src/helpers/get-app.ts]: hmm i get what you mean, but it seems like on the frontend, there isn't much logic for the labels also? not sure if it was a bad practice with the initial code

File: packages/frontend/src/components/FlowStep/index.tsx [hunk 4]
```diff
@@ -301,6 +303,7 @@ export default function FlowStep(
                       onSubmit={expandNextStep}
                       onChange={handleChange}
                       step={step}
+                      settingsLabel={selectedActionOrTrigger?.settingsStepLabel}
```
Comment (ogp-weeloong) [inline packages/frontend/src/components/FlowStep/index.tsx]:
```suggestion
                      settingsLabel={selectedActionOrTrigger?.settingsStepLabel ?? app?.settingsStepLabel}
```

**Review by ogp-weeloong — APPROVED** [review]
gogogogoggo

File: packages/frontend/src/components/FlowStep/index.tsx (hunk 4 above)
Comment (m0nggh) [reply packages/frontend/src/components/FlowStep/index.tsx]: ooo okais thks for the catch
